pkg/cluster/cluster.go

```diff
@@ -725,6 +735,31 @@ func (c *Cluster) initInfrastructureRoles() error {
 	return nil
 }
 
+func (c *Cluster) mergeRoleDefinitions(roleA, roleB *spec.PgUser) (*spec.PgUser, error) {
+	// human roles always win
+	// infrastructure roles take precendence over the manifest roles
+	if roleA.Name != roleB.Name {
+		return nil, fmt.Errorf("could not merge role %#v with role %#v: role names don't match")
+	}
+	for _, origin := range []spec.RoleOrigin{spec.RoleOriginSystem, spec.RoleOriginTeamsAPI} {
+		if roleA.Origin == origin {
```

Comment:
do we warn users somewhere if they define roles that coincidentally have the same name as one the system roles ?